guppylang/prelude/_internal/compiler/quantum.py

```diff
@@ -81,11 +81,11 @@ def compile_with_inouts(self, args: list[Wire]) -> CallReturnWires:
         with conditional.add_case(1) as case:
             case.set_outputs(*case.inputs())
 
-        q = self.builder.add_op(
+        qs = self.builder.add_op(
             quantum_op(self.opname)(
-                ht.FunctionType([ht.Qubit, ROTATION_T], [ht.Qubit]), []
+                ht.FunctionType([ht.Qubit for _ in qs] + [ROTATION_T], [ht.Qubit for _ in qs]), []
             ),
-            q,
+            *qs,
             conditional,
         )
-        return CallReturnWires(regular_returns=[], inout_returns=[q])
+        return CallReturnWires(regular_returns=[], inout_returns=qs)
```

Comment:
```suggestion
        return CallReturnWires(regular_returns=[], inout_returns=list(*qs))
```

Something like this should fix the mypy error

Reply:
This gets me:
```
>       return CallReturnWires(regular_returns=[], inout_returns=list(*qs))
E       TypeError: 'OutPort' object is not iterable
```

Note I am assigning qs to the return of self.builder.add_op - I probably need advice on that.

Reply:
Oops I think it should be `list(qs)` then